room/compiler-processing/src/test/java/androidx/room/compiler/processing/XRoundEnvTest.kt

```diff
@@ -0,0 +1,113 @@
+import androidx.room.compiler.processing.util.Source
+import androidx.room.compiler.processing.util.getDeclaredMethod
+import androidx.room.compiler.processing.util.getField
+import androidx.room.compiler.processing.util.getMethod
+import androidx.room.compiler.processing.util.runProcessorTest
+import com.google.common.truth.Truth
+import org.junit.Test
+
+class XRoundEnvTest {
+
+    @Test
+    fun getAnnotatedElements() {
+        val source = Source.kotlin(
+            "Baz.kt",
+            """
+            import androidx.room.compiler.processing.testcode.OtherAnnotation;
+            @OtherAnnotation(value="xx")
+            class Baz {
+                @OtherAnnotation(value="xx")
+                var myProperty: Int = 0
+                
+                @OtherAnnotation(value="xx")
+                fun myFunction() { }
+            }
+            """.trimIndent()
+        )
+
+        runProcessorTest(listOf(source)) { testInvocation ->
+            val annotatedElementsByClass = testInvocation.roundEnv.getElementsAnnotatedWith(
+                OtherAnnotation::class
+            )
+
+            val annotatedElementsByName = testInvocation.roundEnv.getElementsAnnotatedWith(
+                OtherAnnotation::class.qualifiedName!!
+            )
+
+            val targetElement = testInvocation.processingEnv.requireTypeElement("Baz")
+
+            Truth.assertThat(
+                annotatedElementsByClass
+            ).apply {
+                containsExactlyElementsIn(annotatedElementsByName)
+                hasSize(3)
+                contains(targetElement)
+                contains(targetElement.getMethod("myFunction"))
+
+                if (testInvocation.isKsp) {
+                    contains(targetElement.getField("myProperty"))
+                } else {
+                    // Javac sees a property annotation on the synthetic function
+                    contains(targetElement.getDeclaredMethod("getMyProperty\$annotations"))
+                }
+            }
+        }
+    }
+
+    @Test
+    fun getAnnotatedPropertyElements() {
+        val source = Source.kotlin(
+            "Baz.kt",
+            """
+            import androidx.room.compiler.processing.testcode.OtherAnnotation;
+            class Baz {
+                @get:OtherAnnotation(value="xx")
+                var myProperty1: Int = 0
+
+                @set:OtherAnnotation(value="xx")
+                var myProperty2: Int = 0
+                
+                @field:OtherAnnotation(value="xx")
+                var myProperty3: Int = 0
+            }
+            """.trimIndent()
+        )
+
+        runProcessorTest(listOf(source)) { testInvocation ->
+            val annotatedElements = testInvocation.roundEnv.getElementsAnnotatedWith(
+                OtherAnnotation::class
+            )
+
+            val targetElement = testInvocation.processingEnv.requireTypeElement("Baz")
+
+            Truth.assertThat(
```

Comment:
nit: we use static import for Truth (though i wouldn't be surprised if there are other instances in the codebase :) ).